packages/react/src/internal/useResizeObserver.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useRef, useState, useLayoutEffect, useEffect } from 'react';
+
+export const useResizeObserver = ({ ref, onResize, deps = [] }) => {
+  const [width, setWidth] = useState(-1);
+  const [height, setHeight] = useState(-1);
+  const entriesToHandle = useRef<ResizeObserverEntry[] | null>(null);
+  const cb = useRef(onResize);
+
+  useEffect(() => {
+    // ref for onResize removes it as dependency from useLayoutEffect
+    // This significantly reduces repeated calls if a function is redefined on every
+    // render
+    cb.current = onResize;
+  }, [onResize]);
+
+  useEffect(() => {
+    const getInitialSize = () => {
+      if (ref.current) {
+        const refComputedStyle = window.getComputedStyle(ref.current);
+        const initialWidth =
+          (ref.current?.offsetWidth || 0) -
+          (typeof refComputedStyle?.paddingLeft === 'string' &&
```

Comment:
I wondered here about logical property support but it looks like this is taken care of by the [spec](https://www.w3.org/TR/css-logical-1/#:~:text=APIs%20that%20return%20computed%20values%20(such%20as%20getComputedStyle())%20must%20return%20the%20same%20value%20for%20equivalent%20pairs%20of%20such%20properties.)

> APIs that return computed values (such as `getComputedStyle()`) must return the same value for equivalent pairs of such properties